components/brave_new_tab_ui/hooks/PrefHookManager.ts

```diff
@@ -27,7 +27,7 @@ export class PrefHookManager<T> {
         getInitialValue().then(this.notifyListeners)
     }
 
-    notifyListeners (prefs: T) {
+    notifyListeners = (prefs: T) => {
```

Comment:
Whoa, it is my understanding that, when this was called by `getInitialValue()`, `this` wasn't bound and caused failure. And you fixed this by using arrow function as `this` in arrow function is who defines it, in this case `PrefHookMananger<t>`. Am I right? Interesting!

So do you usually prefer this methodology than `bind`? To me, using arrow function looks more handy and robust.

Reply:
Yeah, exactly :smile: Generally, I prefer arrow functions because they're easier to reason about but they do have a couple of downsides when used with classes. For example, if you use an arrow function on a class, each instance of the class will have a separate function, whereas with a traditional JS function they'll be the same instance.

```ts
class FuncClass {
    myFunc() {}
}

class ArrowClass {
    myFunc = () => {}
}

console.log(new FuncClass().myFunc === new FuncClass().myFunc) // true
console.log(new ArrowClass().myFunc === new ArrowClass().myFunc) // false

```

but if we bind we get the same result anyway :laughing: